requirements-dev.txt

```diff
@@ -1,3 +1,4 @@
+assertpy
 coverage
 mock
-nose
\ No newline at end of file
+pytest
```

Comment:
The test changes look good! I'm just wondering why do you prefer pytest over unittest in this case. I don't have much experience with pytest and would be nice to know your reason in case I use it in my future projects :-)

Reply:
pytest has simpler assertions and many tools. Also, one can use simple test methods and fixtures, instead of the class structure imposed by unittest.